Corona-Warn-App/src/main/java/de/rki/coronawarnapp/service/submission/ScanResult.kt

```diff
@@ -0,0 +1,34 @@
+package de.rki.coronawarnapp.service.submission
+
+import java.util.regex.Pattern
+
+data class ScanResult(val rawResult: String) {
+    val isValid by lazy {
+        containsValidGUID(rawResult)
+    }
+    val guid: String? by lazy {
+        if (isValid) guid
+        else null
+    }
+
+    private fun matchGUIDPattern(scanResult: String): Boolean {
+        val pattern = QR_CODE_REGEX
+        return pattern.toRegex().matches(scanResult)
+    }
+
+    private fun containsValidGUID(rawResult: String): Boolean {
+        if (rawResult.length > SubmissionConstants.MAX_QR_CODE_LENGTH ||
+            rawResult.count { it == SubmissionConstants.GUID_SEPARATOR } != 1 ||
+            !matchGUIDPattern(rawResult)
+        )
+            return false
+
+        val potentialGUID = SubmissionService.extractGUID(rawResult)
```

Comment:
Also move this into this class, all other instances where someone needs to call `extractGUID` they should work with the `ScanResult` class.